torchbenchmark/models/sam_fast/__init__.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# This software may be used and distributed according to the terms of the GNU General Public License version 3.
+
+from ...util.model import BenchmarkModel
+from segment_anything_fast.build_sam import sam_model_fast_registry
+from segment_anything_fast.predictor import SamPredictor
+import numpy as np
+import cv2
+from torchbenchmark.tasks import COMPUTER_VISION
+import torch
+import os
+
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.SEGMENTATION
+    DEFAULT_EVAL_BSIZE = 32
+
+    def __init__(self, test, device, batch_size=1, extra_args=[]):
+        super().__init__(
+            test=test, device=device, batch_size=batch_size, extra_args=extra_args
+        )
+
+        # Checkpoint options are here https://github.com/facebookresearch/segment-anything#model-checkpoints
+        data_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), ".data")
+        sam_checkpoint = os.path.join(data_folder, "sam_vit_h_4b8939.pth")
+        model_type = "vit_h"
+
+        self.model = sam_model_fast_registry[model_type](checkpoint=sam_checkpoint)
+        self.model.to(device=device)
+        data_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), ".data")
+
+        image_path = os.path.join(data_folder, "truck.jpg")
+        self.image = cv2.imread(image_path)
+        self.image = cv2.cvtColor(self.image, cv2.COLOR_BGR2RGB)
+        self.sample_image = torch.randn((3, 256, 256)).to(device)
+
+    def get_module(self):
+        example_input = [
+            {
+                "image": self.sample_image,
+                "original_size": (256, 256),
+                "point_coords": torch.tensor([[[1,1]]], device=self.device),
```

Comment:
Had to add some dummy point coords because segment_anything_fast actually requires at least one of point_coords, boxes, or masks. Otherwise, this line will fail since sparse_embeddings is undefined otherwise:
https://github.com/pytorch-labs/segment-anything-fast/blob/main/segment_anything_fast/modeling/prompt_encoder.py#L170

This seems like a bug since normal sam doesn't have this requirement. But working around for now.